handle_multi_pay_invoice_request.go

```diff
@@ -0,0 +1,197 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+	"sync"
+
+	"github.com/nbd-wtf/go-nostr"
+	decodepay "github.com/nbd-wtf/ln-decodepay"
+	"github.com/sirupsen/logrus"
+)
+
+func (svc *Service) HandleMultiPayInvoiceEvent(ctx context.Context, sub *nostr.Subscription, request *Nip47Request, event *nostr.Event, app App, ss []byte) {
+
+	nostrEvent := NostrEvent{App: app, NostrId: event.ID, Content: event.Content, State: "received"}
+	err := svc.db.Create(&nostrEvent).Error
+	if err != nil {
+		svc.Logger.WithFields(logrus.Fields{
+			"eventId":   event.ID,
+			"eventKind": event.Kind,
+			"appId":     app.ID,
+		}).Errorf("Failed to save nostr event: %v", err)
+		svc.Logger.WithFields(logrus.Fields{
+			"eventId":   event.ID,
+			"eventKind": event.Kind,
+		}).Errorf("Failed to process event: %v", err)
+		return
+	}
+
+	multiPayParams := &Nip47MultiPayParams{}
+	err = json.Unmarshal(request.Params, multiPayParams)
+	if err != nil {
+		svc.Logger.WithFields(logrus.Fields{
+			"eventId":   event.ID,
+			"eventKind": event.Kind,
+			"appId":     app.ID,
+		}).Errorf("Failed to decode nostr event: %v", err)
+		svc.Logger.WithFields(logrus.Fields{
+			"eventId":   event.ID,
+			"eventKind": event.Kind,
+		}).Errorf("Failed to process event: %v", err)
+		return
+	}
+
+	var wg sync.WaitGroup
+	for _, invoiceInfo := range multiPayParams.Invoices {
+		wg.Add(1)
+		go func(invoiceInfo InvoiceInfo) {
+			defer wg.Done()
+			bolt11 := invoiceInfo.Invoice
+			// Convert invoice to lowercase string
+			bolt11 = strings.ToLower(bolt11)
+			paymentRequest, err := decodepay.Decodepay(bolt11)
+			if err != nil {
+				svc.Logger.WithFields(logrus.Fields{
+					"eventId":   event.ID,
+					"eventKind": event.Kind,
+					"appId":     app.ID,
+					"bolt11":    bolt11,
+				}).Errorf("Failed to decode bolt11 invoice: %v", err)
+
+				resp, err := svc.createResponse(event, Nip47Response{
+					ResultType: NIP_47_PAY_INVOICE_METHOD,
```

Comment:
this should be `MULTI_`, right?

we should include the tag, but it's impossible if the invoice was failed to be decoded. Hmm...

Reply:
This is an edge case not considered by the spec. What should we do here?

Reply:
To be clear: from the spec we need to add the payment hash as the d tag if no id is provided. But if we can't decode the invoice then we can't get the payment hash, and then we can't show what invoice was failed to be sent.

Reply:
Decided to ignore this for now.